src/mbgl/map/tile_worker.cpp

```diff
@@ -87,21 +113,15 @@ void TileWorker::parseLayer(const StyleLayer& layer, const GeometryTile& geometr
         return;
     }
 
-    // This is a singular layer. Check if this bucket already exists.
-    if (getBucket(layer))
```

Comment:
What's happening here? I believe this check is what makes the layer `ref` mechanism efficient, by avoiding creating duplicate buckets for the referent layers.